plugins/notion/src/data.ts

```diff
@@ -401,6 +412,40 @@ export function getFieldDataEntryForProperty(
             return { type: "multiCollectionReference", value: property.relation.map(({ id }) => id) }
         }
         case "files": {
+            if (field.type === "array") {
+                const imageField = field.fields[0]
+
+                return {
+                    type: "array",
+                    value: property.files
+                        .map((file): ArrayItemInput | undefined => {
+                            switch (file?.type) {
```

Comment:
nit: could we add an exhaustive check on the file type and filter out the undefined values first?

Reply:
Made it exhaustive. Can't filter first without making the code uglier